### PR TITLE
WIP: Add tests and update params logic

### PR DESCRIPTION
**Pull request by flynneva**
- add more tests, fix bugs found
- reorganize parameters logic

### CODE REVIEW

File: include/usb_cam/usb_cam.hpp [hunk 1]
```diff
@@ -314,6 +320,79 @@ class UsbCam
     return m_image.pixel_format;
   }
 
+  /// @brief Send current parameters to V4L2 device
+  /// TODO(flynneva): only send parameters that changed
+  inline void set_v4l2_params()
```
Comment (twdragon) [inline include/usb_cam/usb_cam.hpp:325]: Are the inline functions really needed here?  Maybe, it is better to implement the classic prototype + body model? For the function called only once it is not ever needed, why we should keep this [useless] specifier here? I guess, it is inherited from the legacy and we can remove the `inline` specifiers without any problem. This is only my opinion, so please share what you think about any real reason we have here

File: test/test_usb_cam_lib.cpp [hunk 2]
```diff
@@ -35,25 +35,86 @@
...
+  // Test to make sure a few parameters were initilized properly
+  // and they can be accessed directly if needed
+  ASSERT_EQ(m_test_parameters.camera_name, "test_camera");
+  ASSERT_EQ(m_test_parameters.device_name, "/dev/video0");
+  ASSERT_EQ(m_test_parameters.frame_id, "test_camera_frame");
+}
+
+TEST_F(test_usb_cam_lib_fixture, usb_cam_class_basic) {
+  // Ensure parameters were properly set using some helper functions
+  ASSERT_EQ(m_test_cam->get_image_width(), size_t(480));
```
Comment (twdragon) [inline test/test_usb_cam_lib.cpp:96]: 480x640 or 640x480? Is the frame orientation inverted?

**Comment by twdragon** [comment]
https://github.com/ros-drivers/usb_cam/actions/runs/4657295606/jobs/8245890280?pr=242#step:5:1883

```sh
Run cd /home/runner/work/_actions/flynneva/renode-linux-runner-action/disable-multicore-emulation
  cd /home/runner/work/_actions/flynneva/renode-linux-runner-action/disable-multicore-emulation
  sudo python3 action/run-in-renode.py "ls /dev | grep video
  " "vivid"
  shell: bash --noprofile --norc -e -o pipefail {0}
python3: can't open file 'action/run-in-renode.py': [Errno 2] No such file
```

@flynneva it seems one script from the internal CI setup is missed

**Comment by flynneva** [comment]
@twdragon yeah this is still a WIP....sorry I forgot to mention that

**Comment by flynneva** [comment]
@twdragon im close to getting a virtual V4l2 device up and running in CI, meaning we can do more integration testing on each PR

**Comment by flynneva** [comment]
Going to close this MR until renode supports more architecdtures. Otherwise we'd have to recompile ROS 2 for RISC-V 🙃

**Comment by mgielda** [comment]
Hi @flynneva note that Renode does support many architectures, including wide support for ARM (see some updates in https://antmicro.com/blog/2023/09/renode-1-14-release/). Of course we need to update the action to feature all the new archs, but armv7-a is possible now, and armv8-a is coming soon (and we can make this "sooner" if there is a need!)

See: https://github.com/antmicro/renode-linux-runner-action#architecture

**Comment by mgielda** [comment]
Please also shoot me an email at mgielda@antmicro.com, happy to chat more!

**Pull request closed without merging**